### DISCUSSION TITLE
Why does esmodule-helpers make property not configurable?

### DISCUSSION THREAD

**Discussion by brantfengfoxtel** [parcel-bundler/parcel#8512] [Q&A]
I am using sinonjs and mocha in my repo. Before I run mocha, I use parcel to package the source code.
The interesting thing is that sinon will throw the following error. It seems parcel doesn't work well with sinon together.
![image](https://user-images.githubusercontent.com/106954273/192437175-0e4604f7-5405-41ab-9ca0-f925f64def99.png)
Here is a demo to reproduce the issue https://github.com/brantfengfoxtel/spike-sinon/blob/main/test/index.js.

Finally, I find that parcel doesn't add configurable: true when exporting es6 modules. https://github.com/parcel-bundler/parcel/blob/v2/packages/transformers/js/src/esmodule-helpers.js#L26

My question is why does esmodule-helpers make property not configurable? Is this by design?

**Comment by brantfengfoxtel** [comment]
Any answer to this question?

**Reply by mischnic** [reply]
I guess since native ESM (e.g. in Node) also doesn't allow this:
```
import * as y from "./y.mjs"; // contains `export const foo = 2;`

Object.defineProperty(y, "foo", {
	enumerable: true,
	value: 3,
});
```
fails with
```
Object.defineProperty(y, "foo", {
       ^

TypeError: Cannot assign to read only property 'foo' of object '[object Module]'
```

Does this work with Webpack (or whatever tool you'd use here instead of Parcel)?

**Reply by brantfengfoxtel** [reply]
@mischnic 
I tried webpack. It doesn't add configurable: true either.
You are right, it is the rule of native ESM.
